docs/aro-hcp-debug-cheatsheet.md

```diff
@@ -0,0 +1,245 @@
+
+# ARO HCP Debugging Cheatsheet
+
+This document is a cheatsheet for debugging ARO HCP cluster creation issues. It shows the rough flow of the cluster creation process through the individual service components. Each component is described in a separate section, including how to access it and how to query for the relevant resources.
+
+>[!NOTE]
+> This document it neither a complete overview of the ARO HCP architecture nor a comprehensive SOP/TSG for debugging during oncall. It is meant as a quick reference and should be considered a learning resource. PTAL at the respective SOPs/TSGs for all platform and service components (once they are available).
+
+## Cluster Creation Flow in a Nutshell
+
+```mermaid
+---
+config:
+  layout: dagre
+---
+flowchart TB
+ arm["ARM"]
+subgraph svc_cluster["service cluster"]
+    subgraph istio_ns["aks-istio-ingress"]
+        istio_ingress["Istio Ingress"]
+    end
+    subgraph mise_ns["ns mise"]
+        mise["MISE"]
+    end
+    subgraph rp_ns["ns aro-hcp"]
+        rp_backend["RP Backend"]
+        rp_frontend["RP Frontend"]
+    end
+    subgraph cs_ns["ns cluster-service"]
+        cs["Clusters Service"]
+    end
+    subgraph maestro_server_ns["ns maestro"]
+        maestro_server["Maestro Server"]
+    end
+ end
+ subgraph mgmt_cluster["management cluster"]
+    subgraph maestro_agent_ns["ns maestro"]
+        maestro_agent["Maestro Agent"]
+    end
+    subgraph acm_mce_ns["ns multicluster-engine"]
+        acm["ACM MCE"]
+    end
+    subgraph hypershift_ns["ns hypershift"]
+        hypershift["Hypershift Operator"]
+    end
+    subgraph hypershift_hcp_ns["ns ocm-xxx-$cluster-id"]
+        hostedcluster_cr["HostedCluster CR"]
+    end
+    subgraph hypershift_cp_ns["ns ocm-xxx-$cluster-id-$name"]
```

Comment:
nit: HostedCluster is projected into HostedControlPlane before creating these pods